sklearn/linear_model/tests/test_ransac.py

```diff
@@ -90,12 +87,13 @@ def test_ransac_max_trials():
                                        random_state=0)
     assert_raises(ValueError, ransac_estimator.fit, X, y)
 
-    ransac_estimator = RANSACRegressor(base_estimator, min_samples=2,
-                                       residual_threshold=5, max_trials=11,
-                                       random_state=0)
-    assert getattr(ransac_estimator, 'n_trials_', None) is None
-    ransac_estimator.fit(X, y)
-    assert_equal(ransac_estimator.n_trials_, 2)
+    for i in range(400):
+        ransac_estimator = RANSACRegressor(base_estimator, min_samples=2,
+                                           random_state=i)
+        assert getattr(ransac_estimator, 'n_trials_', None) is None
+        ransac_estimator.fit(X, y)
+        max_trials = _dynamic_max_trials(len(X) - len(outliers), X.shape[0], X.shape[1] + 1, 1 - 1e9)
```

Comment:
1e9 is a much smaller number than 1-ransac_estimator.stop_probability. Using the same stop_probability for estimation and varification would cause a non-trivial chance of failure even for well behaved code. I could use (1-ransac_estimator.stop_probability)*1e3. The point is that is should be much smaller